mslib/msui/mscolab.py

```diff
@@ -317,77 +323,30 @@ def login_handler(self):
 
         if r.text == "False":
             # show status indicating about wrong credentials
-            self.set_status("Error", 'Oh no, you need to add a user account or '
+            self.set_status("Error", 'Wrong username or password. '
                             f'<a href="{url_recover_password}">Recover Your Password</a>')
-        elif r.text == "Unauthorized Access":
-            # Server auth required for logging in
-            self.login_data = [data, r, url]
-            self.connectBtn.setEnabled(False)
-            self.stackedWidget.setCurrentWidget(self.httpAuthPage)
-            try:
-                self.httpBb.accepted.disconnect()
-            except TypeError:
-                pass
-            try:
-                self.httpBb.rejected.disconnect()
-            except TypeError:
-                pass
-            self.httpBb.accepted.connect(self.login_server_auth)
-            self.httpBb.rejected.connect(lambda: self.stackedWidget.setCurrentWidget(self.loginPage))
         else:
-            try:
-                save_password_to_keyring(service_name="MSCOLAB", username=emailid, password=password)
-            except (NoKeyringError, PasswordSetError, InitError) as ex:
-                logging.warning("Can't use Keyring on your system: %s" % ex)
-            self.mscolab.after_login(emailid, self.mscolab_server_url, r)
+            self.save_user_credentials_to_config_file(data["email"], data["password"])
+            self.mscolab.after_login(data["email"], self.mscolab_server_url, r)
 
     def save_user_credentials_to_config_file(self, emailid, password):
-        data_to_save_in_config_file = {
-            "MSCOLAB_mailid": emailid
-        }
-
         try:
-            save_password_to_keyring(service_name="MSCOLAB", username=emailid, password=password)
+            save_password_to_keyring(service_name=self.mscolab_server_url, username=emailid, password=password)
         except (NoKeyringError, PasswordSetError, InitError) as ex:
             logging.warning("Can't use Keyring on your system:  %s" % ex)
-        exiting_mscolab_mailid = config_loader(dataset="MSCOLAB_mailid")
-        if exiting_mscolab_mailid != emailid:
+        http_auth = config_loader(dataset="MSS_auth")
+        if http_auth.get(self.mscolab_server_url) != emailid:
```

Comment:
http_auth is not based on an emailid, the users name can be any name.  I think that was destinguished before in auth_user and auth_password, or?

at least the docu has a hint to that https://mss.readthedocs.io/en/stable/usage.html#mscolab-login-and-www-authentication

Reply:
renamed to fix problem.